src/main/java/org/osiam/addons/self_administration/registration/RegistrationController.java

```diff
@@ -92,9 +94,11 @@ public String register(@Valid final RegistrationUser registrationUser,
             return "registration";
         }
 
-        user = registrationService.saveRegistrationUser(user);
-
-        registrationService.sendRegistrationEmail(user, request);
+        try {
+            registerUserOrRollback(user, request);
+        } catch (RollbackUserCreationException e) {
+            return "registration";
```

Comment:
An error message should be returned, like [here](https://github.com/sputnik27/addon-self-administration/blob/rollback-user-creation-after-sending-mail-failed/src/main/java/org/osiam/addons/self_administration/registration/RegistrationController.java#L91).